app/models/setting.rb

```diff
@@ -104,11 +105,14 @@ def self.method_missing(method, *args)
 
   def value=(v)
     v = v.to_yaml unless v.nil?
+    # the has_attribute is for enabling DB migrations on older versions
+    v = encrypt_field(v) if has_attribute?(:encrypted) && encrypted
     write_attribute :value, v
   end
 
   def value
     v = read_attribute(:value)
+    v = decrypt_field(v) if has_attribute?(:encrypted) && encrypted
```

Comment:
I'd suggest leaving the conditional out here, so it's possible to change encrypted from true to false and still decrypt existing values - simply decrypt automatically when possible.